src/analysis/jump.ml

```diff
@@ -127,19 +127,54 @@ let rec skip_non_moving pos = function
   | [] -> []
 ;;
 
+let rec find_case_pos pos cases cmp_fn =
+  match cases with
+  | [] -> None
+  | { c_lhs = { pat_loc; _ }; _ } :: tail ->
+    let case_loc = pat_loc.loc_start in
+    if cmp_fn pos case_loc then
+      Some case_loc
```

Comment:
Correct me if I am wrong, but I am under the impression that since you compare the cases from the beginning every time when trying to jump to the previous case you will always jump to the first one. This is not part of your tests:: try to jump from case ` | 'O' ->` to the previous one.

Reply:
If we try to jump from the first case to a previous case it should return an exception saying there's no previous case.

Reply:
You asked for another review but this comment does not appear to be addressed, did you forgot to push some changes?

Reply:
Thanks for catching that.

Reply:
I think all commits are now pushed.